Tests/MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -563,6 +595,24 @@ extension NavigationViewControllerTests: NavigationViewControllerDelegate, Style
     func navigationViewController(_ navigationViewController: NavigationViewController, roadNameAt location: CLLocation) -> String? {
         return customRoadName[location.coordinate] ?? nil
     }
+    
+    func navigationViewController(_ navigationViewController: NavigationViewController, willAddRouteLineLayer layer: LineLayer, identifier: String) -> LineLayer? {
+        var lineLayer = layer
+        lineLayer.lineOpacity = .constant(expectedLineOpacity)
```

Comment:
Can we use in a mocked delegate the different values for RouteLineLayer, RouteCasingLineLayer, RouteRestrictedAreas?

Now we do not test if those delegates methods not get messed up and e.g.  the controller might set layer for RouteCasingLineLayer as RouteRestrictedAreas and test will still pass

Reply:
Thanks for the suggestion, the mock delegate is added for this specific test and different values have been assigned to different layers.